analyzers/src/SonarAnalyzer.Common/Extensions/ComparisonExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+using Comparison = SonarAnalyzer.Helpers.ComparisonKind;
+namespace SonarAnalyzer.Helpers
+{
+    public static class ComparisonKindExtensions
+    {
+        public static Comparison ComparisonKind(this IMethodSymbol method) =>
+            method?.Name switch
```

Comment:
Let's check for `MethodKind == MethodKind.UserDefinedOperator` first. Some joker can declare a method like `public void op_LessThan() { }` that is not a comparison operator.